pkgs/by-name/an/angular-language-server/package.nix

```
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  fetchYarnDeps,
+  fetchFromGitHub,
+  nodejs,
+  yarn,
+  bazel,
+  yarnConfigHook,
+  yarnInstallHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vscode-ng-language-service";
+  version = "18.2.0";
+
+  src = fetchFromGitHub {
+    owner = "angular";
+    repo = "vscode-ng-language-service";
+    rev = "v${version}";
+    hash = "sha256-9+WWKvy/Vu4k0BzJwPEme+9+eDPI1QP0+Ds1CbErCN8=";
+  };
+
+  yarnLock = src + "/yarn.lock";
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    hash = "sha256-N0N0XbNQRN7SHkilzo/xNlmn9U/T/WL5x8ttTqUmXl0=";
+  };
+
+  nativeBuildInputs = [
+    yarn
+    bazel
+    yarnConfigHook
+    yarnInstallHook
+  ];
+
+  buildInputs = [
+    nodejs
+    bazel
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    tsc -b && yarn bazel build :npm
+
+    runHook postBuild
+  '';
```

```suggestion
  yarnBuildScript = "compile";
```

https://github.com/angular/vscode-ng-language-service/blob/1e4c92c156c0f1cdd2c9510000a6f41f070dd123/package.json#L224C6-L224C13

Assuming this would work.

---

Oddly, using the yarnBuildHook and package.json script directly results in yarn attempting to install bazel on my machine. Explicitly doing the command instead of the equivalent script does not.

---

That's because bazel isn't even run:
```
/nix/store/91d27rjqlhkzx7mhzxrir1jcr40nyc7p-stdenv-linux/setup: line 1709: tsc: command not found
```